services/horizon/internal/actions/helpers.go

```
@@ -701,3 +760,62 @@ func FullURL(ctx context.Context) *url.URL {
 	}
 	return url
 }
+
+// Note from chi: it is a good idea to set a Decoder instance as a package
+// global, because it caches meta-data about structs, and an instance can be
+// shared safely:
+var decoder = schema.NewDecoder()
+
+// GetParams fills a struct with values read from a request's query parameters.
+func GetParams(dst interface{}, r *http.Request) error {
+	query := r.URL.Query()
+
+	// Merge chi's URLParams with URL Query Params. Given
+	// `/accounts/{account_id}/transactions?foo=bar`, chi's URLParams will
+	// contain `account_id` and URL Query params will contain `foo`.
+	if rctx := chi.RouteContext(r.Context()); rctx != nil {
+		for _, key := range rctx.URLParams.Keys {
+			query.Set(key, rctx.URLParam(key))
```

should we return an error if a route param collides with an existing query param?

---

sounds like a good idea, we should probably panic in that scenario

---

@tamirms I was thinking about this and if there is a collision, I think we should give priority to chi's query params and ignore the passed-in value since it isn't probably one of the defined (expected) query params.

If we define a route like `/accounts/{account_id}/offers`, it's likely we won't have an `account_id` as one of the possible query params, if the request contains it, then we just ignore it.